src/backend/src/utils/reimbursement-requests.utils.ts

```diff
@@ -59,35 +65,44 @@ export const validateReimbursementProducts = async (
   /**
    * Aggregate all the wbsNums for all the reimbursement products
    */
-  const wbsNums = reimbursementProductCreateArgs.map((reimbursementProductInfo) => reimbursementProductInfo.wbsNum);
+  const reasons = reimbursementProductCreateArgs.map((reimbursementProductInfo) => reimbursementProductInfo.reason);
 
   /**
    * Goes through each wbsNum and finds the wbs element associated with it
    * Checks if the wbs element exists and is not deleted
    */
-  const reimbursementProductsWithWbsElement: Promise<{
+  const reimbursementProductsWithReason: Promise<{
     name: string;
     cost: number;
-    wbsElementId: number;
-    wbsNum: WbsNumber;
-  }>[] = wbsNums.map(async (wbsNum, index) => {
-    const wbsElement = await prisma.wBS_Element.findFirst({
-      where: {
-        carNumber: wbsNum.carNumber,
-        projectNumber: wbsNum.projectNumber,
-        workPackageNumber: wbsNum.workPackageNumber
-      }
-    });
-    if (!wbsElement) throw new NotFoundException('WBS Element', wbsPipe(wbsNum));
-    if (wbsElement.dateDeleted) throw new DeletedException('WBS Element', wbsPipe(wbsNum));
+    reason: { wbsElementId: number; wbsNum: WbsNumber } | OtherProductReason;
```

Comment:
type this

Reply:
We should make { wbs.... : wbs... } | OtherProductReason a type and just return that instead of oring it everywhere